src/mainIntegrationContext.js

```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const MainIntegrationContext = React.createContext({
+  saveFile: (filePath, data) => {
```

Comment:
@Quiescent im not sure what this means. I haven't tried creating React Context 🥲

Reply:
It's a way to communicate values through components without explicitly passing them through the props of each one.
(In that way, it's a little bit like redux).
Here are the docs: https://reactjs.org/docs/context.html

e.g.
```
const ExampleContext = React.createContext({ someValueWeWantToPassDown })
```

A component must provide the context, and another must consume it.

Here's a consumer:
```
const SomeInnerComponent = ({}) => {
  return (
    <ExampleContext.consumer>
      {({ someValueWeWantToPassDown }) => {
          return (
            <span>Message passod down without props(!): {someValueWeWantToPassDown}</span>
          )
      }}
    </ExampleContext.consumer>
}

And here, we provide the context so that anything inside the tree that the provider surrounds can access the context.
```
const AComponentForDemoPurposes = () => {
  return (
    <ExampleContext.provider value={{someValueWeWantToPassDown: 'hooray!'}}>
      <div>
        <Form>
          ...
        </Form>
        <SomeComponent>
          <SomeInnerComponent />
        </SomeComponent>
      </div>
    </ExampleContext.provider>
}
```